File: objectiveai-rs/src/vector/completions/cache/http.rs
use crate::{HttpClient, HttpError};

pub async fn get_completion_votes(
    client: &HttpClient,
    id: &str, // vector completion ID
) -> Result<super::response::CompletionVotes, HttpError> {
    client
        .send_unary(
            reqwest::Method::GET,
            &format!("vector/completions/{}", id),
            None::<String>,
        )
        .await
}

pub async fn get_cache_vote(
    client: &HttpClient,
    request: &super::request::CacheVoteRequest,
) -> Result<super::response::CacheVote, HttpError> {
    client
        .send_unary(
            reqwest::Method::GET,
            "vector/completions/cache",
            Some(request),
        )
        .await
}

File: objectiveai-js/src/vector/completions/cache/http.ts
import { ObjectiveAI, RequestOptions } from "../../../client";
import { CacheVoteRequest } from "./request";
import { CacheVote } from "./response";

export function retrieve(
  client: ObjectiveAI,
  request: CacheVoteRequest,
  options?: RequestOptions,
): Promise<CacheVote> {
  // Using GET with body for complex request object
  return client.get_unary<CacheVote>(
    "/vector/completions/cache",
    request,
    options,
  );
}
